actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -143,6 +158,25 @@ def length
           routes.length
         end
 
+        def add_url_helper(name, defaults, &block)
+          @custom_helpers << name
+          helper = DirectUrlHelper.new(name, defaults, &block)
+
+          @path_helpers_module.module_eval do
+            define_method(:"#{name}_path") do |*args|
+              options = args.extract_options!
```

Comment:
Is there any benefit to doing this extraction here? Seems like that could be `call`'s problem, where it already does so for nested arrays.